src/hardware/flash.cpp

```diff
@@ -266,9 +266,9 @@ void saveParameters() {
 
     // CAN ID of the motor controller
     #ifdef ENABLE_CAN
-        writeFlash(CAN_ID_INDEX, (float)getCANID());
+        writeFlash(CAN_ID_INDEX, (uint32_t)getCANID());
     #else
-        writeFlash(CAN_ID_INDEX, (float)-1);
+        writeFlash(CAN_ID_INDEX, (uint32_t)0);
```

Comment:
I'd prefer not to use 0, as that is a valid CAN index

Reply:
may be
```
 writeFlash(CAN_ID_INDEX, (uint32_t)-1);
```

Reply:
That's an unsigned integer though, you can't do that. Maybe we could make a read and write for int32_t, but that would add where we could just use the existing float.

Reply:
(uint32_t)(-1) is legal typecast.
(uint8_t)(-1) : convert the signed int “-1” to an unsigned byte (i.e. “255” == 0xFF)
(uint16_t)(-1) : convert the signed int “-1” to an unsigned byte (i.e. “65535” == 0xFFFF)
(uint32_t)(-1) : convert the signed int “-1” to an unsigned byte (i.e. “4294967295” == 0xFFFFffff)
It doesn't cause a problem.